dynawo/sources/Models/CPP/Controls/Voltage/ModelMinMaxMean/DYNModelMinMaxMean.cpp

```diff
@@ -0,0 +1,365 @@
+  }
+
+  // Add whether a bus is connected or not
+  for (std::size_t i=0; i < nbConnectedInputs_; i++) {
+    name.str("");
+    name.clear();
+    name << "running_" << i << "_value";
+    variables.push_back(VariableNativeFactory::createState(name.str(), BOOLEAN));
+  }
+}
+
+void
+ModelMinMaxMean::defineParameters(vector<ParameterModeler>& parameters) {
+  parameters.push_back(ParameterModeler("nbInputs", VAR_TYPE_INT, EXTERNAL_PARAMETER));
+}
+
+void
+ModelMinMaxMean::setSubModelParameters() {
+  nbConnectedInputs_ = findParameterDynamic("nbInputs").getValue<int>();
+}
+
+/**
+ * @brief Reference frequency elements initializer
+ *
+ * Define elements for this model( elements to be seen by other models)
+ *
+ * @param elements  Reference to elements' vector
+ * @param mapElement Map associating each element index in the elements vector to its name
+ */
+void
+ModelMinMaxMean::defineElements(std::vector<Element> &elements, std::map<std::string, int>& mapElement) {
+  addElement("min", Element::STRUCTURE, elements, mapElement);
+  addSubElement("value", "min", Element::TERMINAL, name(), modelType(), elements, mapElement);
+  addElement("max", Element::STRUCTURE, elements, mapElement);
+  addSubElement("value", "max", Element::TERMINAL, name(), modelType(), elements, mapElement);
+  addElement("avg", Element::STRUCTURE, elements, mapElement);
+  addSubElement("value", "avg", Element::TERMINAL, name(), modelType(), elements, mapElement);
+  // Now deal with the continuous variables
+  stringstream names;
+  for (size_t i = 0; i < nbConnectedInputs_; ++i) {
+    names.str("");
+    names.clear();
+    names << "UMonitored_" << i;
+    addElement(names.str(), Element::STRUCTURE, elements, mapElement);
+    addSubElement("value", names.str(), Element::TERMINAL, name(), modelType(), elements, mapElement);
+  }
+  // Now let's handle the boolean parts
+  for (size_t i = 0; i < nbConnectedInputs_; ++i) {
+    names.str("");
+    names.clear();
+    names << "running_" << i;
+    addElement(names.str(), Element::STRUCTURE, elements, mapElement);
+    addSubElement("value", names.str(), Element::TERMINAL, name(), modelType(), elements, mapElement);
+  }
+}
+
+void
+ModelMinMaxMean::dumpUserReadableElementList(const std::string& /*nameElement*/) const {
+  Trace::info() << DYNLog(ElementNames, name(), modelType()) << Trace::endline;
+  Trace::info() << "  ->" << "UMonitored_" << "<0-" << nbConnectedInputs_ << ">_value" << Trace::endline;
+  Trace::info() << "  ->" << "running_" << "<0-" << nbConnectedInputs_ << ">_value" << Trace::endline;
+}
+
+void
+ModelMinMaxMean::setFequations() {
+  // setFequations not needed
+}
+
+void
+ModelMinMaxMean::checkDataCoherence(const double /*t*/) {
+  for (std::size_t i = 0; i < nbConnectedInputs_; ++i) {
+    if (isConnected(i)) {  // Current input is active.
+      if (yLocal_[i] < calculatedVars_[minValIdx_]) {
+        throw DYNError(Error::MODELER, FrequencyCollapse, yLocal_[i], calculatedVars_[minValIdx_]);
```

Comment:
wrong error message

Reply:
Yes indeed. However this data coherency check has been removed, thereby rendering moot any need to define a new error message :)